src/main/java/com/experoinc/janusgraph/diskstorage/foundationdb/FoundationDBTx.java

```diff
@@ -154,7 +151,8 @@ private TransactionClose(String msg) {
         byte[] value = null;
         for (int i = 0; i < maxRuns; i++) {
             try {
-                value = this.tx.get(key).get();
+                ReadTransaction transaction  = getTransaction(this.isolationLevel, this.tx);
```

Comment:
moving reads to another method to leverage snapshot reads.
https://apple.github.io/foundationdb/developer-guide.html#snapshot-reads